fcrepo4/fcrepo4
# Inbound references service

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3166

# What does this Pull Request do?

Adds an API and implementation of a new service to parse incoming RDF for internal references and stores them in a new index, then allows this index to be queried.

# How should this be tested?

1. Create a resource.
1. Create another resource with any reference to the first ie. `<> <http://awoods.com/pointsTo> <http://localhost:8080/rest/firstResource>`
1. Get the first resource, see no inbound reference triples
1. Get the first resource with the header `Prefer: return=representation; include="http://fedora.info/definitions/fcrepo#PreferInboundReferences"`, see the inbound reference.

For binaries when requesting the inbound references on the `fcr:metadata` endpoint you will get both references to the actual binary URI and the `fcr:metadata` URI.

**Note**: I messed up and forgot to remove the references from the index on deletion, I am implementing this now and will make this PR live once it is added.

# Interested parties
@bbpennel  @fcrepo4/committers
